### PR TITLE
Various updates to variables, IAM, commands, docs

### PR DESCRIPTION
**Pull request by mzawadzk**

### CODE REVIEW

**Comment by alexlenk** [comment]
Can you please provide more information on the changes. It's hard to do a peer review based on this comment and 200 changed lines.

File: README.md [hunk 1]
```diff
@@ -5,51 +5,60 @@ This module will create a Kubernetes cluster with [kops](https://github.com/kube
...
 
 
-## Usage:
-### cross-account scenario (deploy from "operations" account into "application"):
+## Usage
+### cross-account scenario (deploy from operations/transit into application):
 ```hcl
 module "kubernetes_cluster_application" {
-  source = "https://github.com/kentrikos/terraform-aws-kops.git"
+  source = "https://github.com/kentrikos/terraform-aws-kops.git"
```
Comment (radepal) [inline README.md]: source = "github.com/kentrikos//terraform-aws-kops"

File: README.md [hunk 2]
```diff
@@ -5,51 +5,60 @@ This module will create a Kubernetes cluster with [kops](https://github.com/kube
...
   iam_cross_account_role_arn = "${var.iam_cross_account_role_arn}"
 }
 ```
 
-### same-account scenario (deploy on "operations" account):
+### same-account scenario (deploy on operations/transit):
 ```hcl
-module "kubernetes_cluster_operations" {
+module "kubernetes_cluster_transit" {
   source = "https://github.com/kentrikos/terraform-aws-kops.git"
```
Comment (radepal) [inline README.md]: source = "github.com/kentrikos//terraform-aws-kops"

File: README.md [hunk 3]
```diff
@@ -5,51 +5,60 @@ This module will create a Kubernetes cluster with [kops](https://github.com/kube
...
-  node_count = "${var.k8s_node_count}"
+  masters_iam_policies_arns  = "${var.k8s_masters_iam_policies_arns}"
+  nodes_iam_policies_arns    = "${var.k8s_nodes_iam_policies_arns}"
 
   iam_cross_account_role_arn = "${var.iam_cross_account_role_arn}"
 }
 ```
 
-### same-account scenario (deploy on "operations" account):
+### same-account scenario (deploy on operations/transit):
```
Comment (radepal) [inline README.md]: Remove "transit"

File: README.md [hunk 4]
```diff
@@ -5,51 +5,60 @@ This module will create a Kubernetes cluster with [kops](https://github.com/kube
...
 
-Currently it must be run from a host that has networking access to VPC where cluster will be deployed (e.g. from kops management node deployed before-hand).
+Currently it must be run from a host that has networking access to VPC where cluster will be deployed (e.g. from Jenkins or kops management node deployed before-hand).
+This module does not create IAM policies on its own (just roles/instance profiles), instead it expect them to be created beforehand and passed as parameters.
 
 
-## Usage:
-### cross-account scenario (deploy from "operations" account into "application"):
+## Usage
+### cross-account scenario (deploy from operations/transit into application):
```
Comment (radepal) [inline README.md]: Remove "transit"

**Comment by mzawadzk** [comment]
> Can you please provide more information on the changes. It's hard to do a peer review based on this comment and 200 changed lines.

* change variables for policies to list of externally provided ARNs             
* rewrite assume-role policy with TF resource instead of here doc               
* cleanup kops command used to deploy/destroy the cluster                       
* fix cluster destroy                                                           
* add variables for master and node instance sizes                              
* updates docs

**Pull request merged**